lib/Driver/ToolChain.cpp

```diff
@@ -280,12 +280,10 @@ mergeBatchInputs(ArrayRef<const Job *> jobs,
   return false;
 }
 
-/// Debugging only: return whether the set of \p jobs is an ordered subsequence
-/// of the sequence of top-level input files in the \c Compilation \p C.
-static bool
-jobsAreSubsequenceOfCompilationInputs(ArrayRef<const Job *> jobs,
-                                      Compilation &C) {
-  llvm::SmallVector<const Job *, 16> sortedJobs;
+static void sortJobsToMatchCompilationInputs(ArrayRef<const Job *> jobs,
+                                             llvm::SmallVectorImpl<const Job *> &sortedJobs,
```

Comment:
Nitpick: no `llvm::`. I'd also consider just having this take an already-populated SmallVectorImpl (or a MutableArrayRef) and sorting it.

Reply:
It's not really a sort (in the sense of a thing you can adapt an in-place sorting algorithm to easily) so much as it's a "build a new vector in the order dictated by the input files list". But I'll fix the names anyways :)

Reply:
Ah, yes, I suppose not.

Reply:
That's a nice function name. Is there a comment anywhere explaining *why* we want the order to match the inputs?